model_fitting/positive_motifs.py

```diff
@@ -0,0 +1,156 @@
+    output_file_statistical = output_path + '_statistical.csv'
+    output_file = output_path + '.csv'
+    df_statistical.to_csv(output_file_statistical, float_format='%.3f')
+    pass_motifs.insert(0, 'label')
+    pass_motifs.insert(0, 'sample_name')
+    df_positive_motifs = df[pass_motifs]
+    df_positive_motifs.to_csv(output_file)
+
+
+def calculation(df, label):    
+    df_mean = pd.DataFrame(df.mean()).transpose()
+    df_std = pd.DataFrame(df.std()).transpose()
+    df_median = pd.DataFrame(df.median()).transpose()
+    df_max = pd.DataFrame(df.max()).transpose()
+    df_min = pd.DataFrame(df.min()).transpose()
+    cal = [f'mean_{label}', f'std_{label}', f'median_{label}', f'max_{label}', f'min_{label}']
+    df_calculation = pd.concat([df_mean, df_std, df_median, df_max, df_min])
+    df_calculation.insert(0, 'label', cal)
+    df_calculation.set_index('label', inplace=True)
+    return df_calculation
+
+def find_positive_motifs(df, threshold_mean, threshold_std, threshold_median, min_max_difference):
+    positive_motifs = []
+    # BC: 0 - mean, 1 - std, 2 - median, 3 - max ,4 -min 
+    # other: 5 - mean, 6 - std, 7 - median, 8 - max, 9 - min
+    for (motif_name, motif_data) in df.iteritems():
+        if ((threshold_mean != 0.0 and  motif_data[0] - motif_data[5] > threshold_mean) or threshold_mean == 0.0) \
+            and ((threshold_std != 0.0 and  motif_data[1] - motif_data[6] > threshold_std) or threshold_std == 0.0) \
+            and ((threshold_median != 0.0 and  motif_data[2] - motif_data[7] > threshold_median) or threshold_median == 0.0) \
+            and (min_max_difference and motif_data[4] > motif_data[8] or not min_max_difference):
+            positive_motifs.append(motif_name)
+    return positive_motifs
+
+
+def statistical_calculation(df, biological_condition, output_path, done_path, threshold_mean, threshold_std, threshold_median,
+                            min_max_difference, rank_method, normalize_factor_hits, normalize_method_hits, normaliza_section, argv):
+    source_df = df.copy()
+    df = df.drop('sample_name', axis=1)
+    df.set_index('label', inplace=True)
+    if rank_method == 'hits':
+        df = normalize(df, normalize_factor_hits, normalize_method_hits, normaliza_section) 
+    if rank_method == 'pval':
+        df = 1-df    
+    df_BC = df.loc[biological_condition]
+    df_other = df.loc['other']
+    # Calculate the statistical functions - mean, std, median 
+    df_BC_statistical = calculation(df_BC, 'BC')
+    df_other_statistical = calculation(df_other, 'other')
+    # Concat two dataframe to one
+    df_statistical = pd.concat([df_BC_statistical, df_other_statistical])
+    # Left only the positive motifs
+    positive_motifs = find_positive_motifs(df_statistical, threshold_mean, threshold_std, threshold_median, min_max_difference)
+    # Write the results 
+    write_results(source_df, df_statistical, positive_motifs, output_path)
```

Comment:
Add to df_statistical the thresholds and column if motif is positive or negative, it's useful for debugging and calibration